ts/identifier_pool.go

```diff
@@ -52,9 +76,9 @@ func (p *identifierPool) GetStringID(ctx context.Context, v string) ID {
 }
 
 // Clone replicates given ID into a new ID from the pool
-func (p *identifierPool) Clone(other ID) ID {
-	id := p.pool.Get().(*id)
-	id.data = other.Data()
+func (p *identifierPool) Clone(v ID) ID {
+	id := p.pool.GetOr(create).(*id)
+	id.pool, id.data = p, append(p.heap.Get(len(v.Data())), v.Data()...)
```

Comment:
Seems faster for some reason, haha:
```go
package main

import "testing"

const size = 4096

func BenchmarkCopy(b *testing.B) {
	src := make([]byte, size)
	dest := make([]byte, size)
	for i := 0; i < b.N; i++ {
		copy(dest, src)
	}
}

func BenchmarkAppend(b *testing.B) {
	src := make([]byte, size)
	dest := make([]byte, size)
	for i := 0; i < b.N; i++ {
		dest = append(dest[:0], src...)
	}
}
```

```
$ go test -v -bench .
testing: warning: no tests to run
BenchmarkCopy-4         20000000               103 ns/op
BenchmarkAppend-4       20000000                67.7 ns/op
PASS
```

Anyway yeah, fine to use append then.